src/settings/tabtexteditor.cpp

```diff
@@ -0,0 +1,155 @@
+#include "tabtexteditor.h"
+
+TabTextEditor::TabTextEditor(QWidget *parent)
+    : QWidget(parent)
+{
+    QHBoxLayout *mainLayout = new QHBoxLayout;
+    mainLayout->addStretch(1);
+
+    // getting data from QSettings
+    QSettings s;
+    styleCurrernt = (s.contains("analizerStyle") ? s.value("analizerStyle").toString() : styleDefault);
+    editorFontSizeCurrernt = (s.contains("analizerFontSize") ? s.value("analizerFontSize").toString() : editorFontSizeDefault);
+    fontCurrent = (s.contains("analizerFontName") ? s.value("analizerFontName").toString() : fontDefault);
+
+    //      STYLE
+    // set a label
+    QLabel *editorStyle = new QLabel(tr(" Editor style"));
+    pcomboEditorStyle = new QComboBox(this);
+    pcomboEditorStyle->addItems(editorStylesList);
+    mainLayout->addWidget(editorStyle);
+    mainLayout->addWidget(pcomboEditorStyle);
+
+    // set combobox acording to settings
+    pcomboEditorStyle->setCurrentIndex(0);
+    for(int i = 0; i < editorStylesList.size(); ++i)
+    {
+        auto &item = editorStylesList[i];
+        if (item == styleCurrernt)
+            pcomboEditorStyle->setCurrentIndex(i);
+    }
+    connect(pcomboEditorStyle,  SIGNAL(currentIndexChanged(const QString&)),
+                                SLOT(onChangeStyle(const QString&)));
+    //      FONT SIZE
+    // set a label
+    QLabel *fontSize = new QLabel(tr(" Font size"));
+    pcomboFontSize = new QComboBox(this);
+    for (int i = 6; i < 25; i += 2)
+        editorFontSizeList << QString::number(i);
+    pcomboFontSize->addItems(editorFontSizeList);
+    mainLayout->addWidget(fontSize);
+    mainLayout->addWidget(pcomboFontSize);
+
+    // set combobox acording to settings
+    pcomboFontSize->setCurrentIndex(0);
+    for(int i = 0; i < editorFontSizeList.size(); ++i)
+    {
+        auto &item = editorFontSizeList[i];
+        if (item == editorFontSizeCurrernt)
+        pcomboFontSize->setCurrentIndex(i);
+    }
+    connect(pcomboFontSize, SIGNAL(currentIndexChanged(const QString&)),
+                            SLOT(onChangeFontSize(const QString&)));
+    //      FONT NAME
+    // handling system fonts
+    setupFontTree();
+    mainLayout->addWidget(fontTree);
+    textEdit = new QTextEdit;                       //  preview font
+    mainLayout->addWidget(textEdit);
+    showFont(fontTree->topLevelItem(fontNumber - 1));
+
+    setLayout(mainLayout);
+
+    connect(fontTree, SIGNAL(currentItemChanged(QTreeWidgetItem*,QTreeWidgetItem*)),
+            this, SLOT(showFont(QTreeWidgetItem*)));
+}
+
+void TabTextEditor::setupFontTree()
+{
+    QFontDatabase database;
+    fontTree = new QTreeWidget;
+    fontTree->setColumnCount(1);
+    fontTree->setHeaderLabels(QStringList() << tr("Font"));
+
+    size_t decentFontNumber = 0;
+    bool foundFont = false;
+    foreach (QString family, database.families()) {
+        const QStringList styles = database.styles(family);
+        if (styles.isEmpty())
+            continue;
```

Comment:
you haven't found any of the style and after that you're trying to find the styles again. I don't understand this part of code. Maybe you meant "break" instead of continue?